boa/src/syntax/lexer/number.rs

```diff
@@ -110,11 +110,52 @@ where
     }
 
     // Consume the decimal digits.
-    cursor.take_while_ascii_pred(buf, &|ch| ch.is_digit(kind.base()))?;
+    take_integer(buf, cursor, kind, true)?;
 
     Ok(())
 }
 
+fn take_integer<R>(
+    buf: &mut Vec<u8>,
+    cursor: &mut Cursor<R>,
+    kind: &NumericKind,
+    separator_allowed: bool,
+) -> Result<(), Error>
+where
+    R: Read,
+{
+    let mut prev_is_underscore = false;
+    let mut pos = cursor.pos();
```

Comment:
Do we need the `pos` variable, cant we call `cursor.pos()` where we need the position?

Reply:
This is to have the correct position when the separator is at the end of the number. If we take `cursor.pos()`, then the position is after the separator, on not on the separator.